kern/gossl_kern.c

```diff
@@ -0,0 +1,90 @@
+/* Copyright © 2022 Hengqi Chen */
+#include "ecapture.h"
+
+struct go_ssl_event {
+	__u64 ts_ns;
+	__u32 pid;
+	__u32 tid;
+	int data_len;
+	char comm[TASK_COMM_LEN];
+	char data[MAX_DATA_SIZE_OPENSSL];
+};
+
+struct {
+	__uint(type, BPF_MAP_TYPE_PERF_EVENT_ARRAY);
+} events SEC(".maps");
+
+struct {
+	__uint(type, BPF_MAP_TYPE_PERCPU_ARRAY);
+	__type(key, __u32);
+	__type(value, struct go_ssl_event);
+	__uint(max_entries, 1);
+} heap SEC(".maps");
+
+static struct go_ssl_event *get_event()
+{
+	static const int zero = 0;
+	struct go_ssl_event *event;
+	__u64 id;
+
+	event = bpf_map_lookup_elem(&heap, &zero);
+	if (!event)
+		return NULL;
+
+	id = bpf_get_current_pid_tgid();
+	event->ts_ns = bpf_ktime_get_ns();
+	event->pid = id >> 32;
+	event->tid = (__u32)id;
+	bpf_get_current_comm(event->comm, sizeof(event->comm));
+	return event;
+}
+
+SEC("uprobe/abi_stack")
+int BPF_KPROBE(probe_stack)
+{
+	struct go_ssl_event *event;
+	__u64 *sp = (void *)ctx->sp, addr;
```

Comment:
please used  `PT_REGS_RET ` macro  instead `ctx->sp`

Reply:
Golang does NOT follow C ABI, define new macros instead.